lib/pipeline.py

```diff
@@ -248,6 +250,28 @@ def generate_fasta(self):
     def split_fasta_basename(self):
         return os.path.join(self.tmp_dir, self.sample_name + "_" + str(self.peptide_sequence_length) + ".fa.split")
 
+    def balance_multithreads(self, iteration_info):
+        for i in range(1,self.n_threads):
+            #find the dimension with the most iterations per thread
+            (max_iterations, dimension) = max(zip(map(lambda x: x['iterations_per_thread'], iteration_info.values()), iteration_info.keys()))
+            #assign that dimension one more thread
+            iteration_info[dimension]['threads'] += 1
+            #calculate how many total threads this combination would use since these are nested threads
+            total_n_threads = iteration_info['file']['threads'] * iteration_info['allele']['threads'] * iteration_info['length']['threads'] * iteration_info['algorithm']['threads']
+            #if total_n_threads exceeds the total number of threads requested, reset to previous state and return
+            if total_n_threads > self.n_threads:
+                iteration_info[dimension]['threads'] -= 1
+                return iteration_info
```

Comment:
Is it possible that adding a thread to one of the other dimensions could be useful and still end up under `n_threads`?

(I was thinking about this using 2, 3, 5, and 8.  `(30)*8 => (30)*9` yields `270` but `2*(120) => 3*(120)` yields `360`.  Not that anyone is running with 300 threads...)

Reply:
Oh yeah, absolutely. I couldn't think of a good way to test though these options without it getting too complicated.